sb_vision/find_3D_coords.py

```diff
@@ -0,0 +1,139 @@
+"""
+Handlers for detecting 3D marker Location.
+
+Also handles loading the calibration file which is passed into the
+location finding function.
+"""
+
+import functools
+import re
+from pathlib import Path
+from typing import List, Tuple, cast
+
+import cv2
+import numpy as np
+from lxml import etree
+
+from sb_vision.coordinates import Cartesian, PixelCoordinate
+
+
+def _get_values_from_xml_element(element: etree.Element) -> List[str]:
+    """Parse an xml tag with space-separated variables."""
+    text = []  # type: List[str]
+    for e in element.itertext():
+        e = e.strip()
+        if e:
+            text += re.split(r'\s+', e)
+    return text
+
+
+def convert(element: etree.Element) -> List[List[np.float64]]:
+    """Converts an element containing an OpenCV matrix to python lists."""
+    type_id = element.attrib.get('type_id')
+    if type_id == 'opencv-matrix':
```

Comment:
Style question: I note that you tend to prefer code along the lines of:
``` python
if condition:
    ...
    return ...
else:
    raise ...
```
rather than:
``` python
if condition:
    raise ...
...
return ...
```
Is there a reason for that?

My preference is for the latter as I find it easier to read & parse (i,e: it has less cognitive load) and like that it results in less nesting (which I find helps make it easy to understand).

Reply:
I think this is a symptom of how I write logic. I prefer to jump into the core logic and work on how to handle edge cases afterwards. If I think of an edge case while writing my code, I usually just stick in an `if` statement to filter away the badness and continue writing. Once the core logic is done, I then think about how to handle edge cases, and the minimal change from there is to just add an `else` with the code to handle them.

I understand the readability benefits of the latter method, so I think I'll try to get into the habit of moving code to the second method once it's written.

Reply:
Another thing is it makes it easier to handle if I wanted to later expand the code to handle more than one condition. In this case I wrote that function with the intention to make it easy to expand to more complex files, hence why in the first version of this, I initially put everything in a dictionary and returned that.

Reply:
Anyways, I've fixed in faa74bd5333d5bea6872e1eb36b7211668d4a8c7